### PR TITLE
feature/extension-jwt-with-user-fields-of-job-add

### PR DESCRIPTION
**Pull request by wwebec**
Extension of JWT implementation of a jobroom2 (gateway application) with fields from a User entity of a  job-ad -service. There were added following fileds into the token:
- extId,
- company, 
- firstName, 
- lastName, 
- email;

### CODE REVIEW

File: src/main/java/ch/admin/seco/jobroom/config/SecurityConfiguration.java [hunk 1]
```diff
@@ -93,7 +99,8 @@ protected void configure(HttpSecurity http) throws Exception {
             .csrf()
             .disable()
             .headers()
-            .cacheControl().disable()
+            .cacheControl()
```
Comment (cjpsycos) [inline src/main/java/ch/admin/seco/jobroom/config/SecurityConfiguration.java]: please reset the formatting in this method

File: src/main/java/ch/admin/seco/jobroom/security/DomainUserDetailsService.java [hunk 2]
```diff
@@ -35,25 +32,12 @@ public DomainUserDetailsService(UserRepository userRepository) {
...
-        List<GrantedAuthority> grantedAuthorities = user.getAuthorities().stream()
-            .map(authority -> new SimpleGrantedAuthority(authority.getName()))
-            .collect(Collectors.toList());
-        return new org.springframework.security.core.userdetails.User(user.getLogin(),
-            user.getPassword(),
-            grantedAuthorities);
+        return userRepository.findOneWithAuthoritiesByLogin(lowercaseLogin)
+                             .map(DomainUserPrincipal::new)
+                             .orElseThrow(() ->
+                                 new UsernameNotFoundException(format(USER_NOT_FOUND_MSG_PTRN, lowercaseLogin)));
```
Comment (cjpsycos) [inline src/main/java/ch/admin/seco/jobroom/security/DomainUserDetailsService.java]: Never give information of non existing users

File: src/main/java/ch/admin/seco/jobroom/security/jwt/TokenResolver.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.Optional;
+
+import javax.servlet.ServletRequest;
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.util.StringUtils;
+
+class TokenResolver {
+
+    static final String TOKEN_PREFIX = "Bearer ";
```
Comment (cjpsycos) [inline src/main/java/ch/admin/seco/jobroom/security/jwt/TokenResolver.java]: Duplication of constants

File: src/main/java/ch/admin/seco/jobroom/security/jwt/ClaimMapper.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                claims.put(lastName.name(), user.getLastName());
+                claims.put(email.name(), user.getEmail());
+            }
+        };
+    }
+
+    static BiConsumer<Claims, Organization> mapOrganisationToClaims() {
+        return (claims, organization) -> {
+            if (organization != null) {
+                claims.put(extId.name(), organization.getExternalId());
```
Comment (cjpsycos) [inline src/main/java/ch/admin/seco/jobroom/security/jwt/ClaimMapper.java]: extId is like the user id in eiam

File: src/main/java/ch/admin/seco/jobroom/security/jwt/ClaimMapper.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+import ch.admin.seco.jobroom.domain.Organization;
+import ch.admin.seco.jobroom.domain.User;
+
+final class ClaimMapper {
+
+    private ClaimMapper() {
+    }
+
+    enum ClaimKey {
+        auth, extId, company, firstName, lastName, email;
```
Comment (cjpsycos) [inline src/main/java/ch/admin/seco/jobroom/security/jwt/ClaimMapper.java]: add userId, phone, companyId
remove company

File: src/main/java/ch/admin/seco/jobroom/security/jwt/ClaimMapper.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+                claims.put(email.name(), user.getEmail());
+            }
+        };
+    }
+
+    static BiConsumer<Claims, Organization> mapOrganisationToClaims() {
+        return (claims, organization) -> {
+            if (organization != null) {
+                claims.put(extId.name(), organization.getExternalId());
+                claims.put(company.name(), organization.getName());
```
Comment (cjpsycos) [inline src/main/java/ch/admin/seco/jobroom/security/jwt/ClaimMapper.java]: claims.put(companyId.name(), organization.getId());

File: src/main/java/ch/admin/seco/jobroom/security/jwt/ClaimMapper.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+            mapUserToClaims().accept(claims, user);
+            mapOrganisationToClaims().accept(claims, user.getOrganization());
+            return claims;
+        };
+    }
+
+    static BiConsumer<Claims, User> mapUserToClaims() {
+        return (claims, user) -> {
+            if (user != null) {
+                claims.put(firstName.name(), user.getFirstName());
```
Comment (cjpsycos) [inline src/main/java/ch/admin/seco/jobroom/security/jwt/ClaimMapper.java]: At the moment:
claims.put(extId.name(), null);

Later with EIAM:
claims.put(extId.name(), user.getExtId());

Comment (wwebec) [reply src/main/java/ch/admin/seco/jobroom/security/jwt/ClaimMapper.java]: done

File: src/main/java/ch/admin/seco/jobroom/security/jwt/ClaimMapper.java (hunk 6 above)
Comment (wwebec) [reply src/main/java/ch/admin/seco/jobroom/security/jwt/ClaimMapper.java]: done

File: src/main/java/ch/admin/seco/jobroom/security/jwt/ClaimMapper.java (hunk 5 above)
Comment (wwebec) [reply src/main/java/ch/admin/seco/jobroom/security/jwt/ClaimMapper.java]: done

File: src/main/java/ch/admin/seco/jobroom/security/jwt/TokenResolver.java (hunk 3 above)
Comment (wwebec) [reply src/main/java/ch/admin/seco/jobroom/security/jwt/TokenResolver.java]: done

**Comment by wwebec** [comment]
The pull request was opened on a branch of fork instead of alv-ch project.
Due to that fact it is closed. The changes are copied to the following pull request
https://github.com/alv-ch/jobroom2/pull/545.

**Pull request closed without merging**